tsdb/engine/tsm1/file_store.go

```diff
@@ -485,12 +491,20 @@ func (f *FileStore) KeyCursor(key string, t int64, ascending bool) *KeyCursor {
 
 func (f *FileStore) Stats() []FileStat {
 	f.mu.RLock()
-	defer f.mu.RUnlock()
+	if len(f.lastFileStats) > 0 {
+		f.mu.RUnlock()
+		return f.lastFileStats
```

Comment:
I think you need to assign this under the read lock and return the assign var or else your accessing the slice outside the lock.